src/main/java/com/direwolf20/buildinggadgets/common/tools/LiquidInteractions.java

```diff
@@ -0,0 +1,82 @@
+package com.direwolf20.buildinggadgets.common.tools;
+
+import com.direwolf20.buildinggadgets.common.utils.helpers.VectorHelper;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceFluidMode;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraftforge.fluids.IFluidBlock;
+
+import java.util.Arrays;
+
+public class LiquidInteractions {
+    public enum GadgetGeneric {
+        IGNORE(0) {
+            public RayTraceResult getLookingAt(EntityPlayer player) {
+                return VectorHelper.getLookingAt(player);
+            }
+        },
+        REPLACE_TOP(1) {
+            public RayTraceResult getLookingAt(EntityPlayer player) {
+                return VectorHelper.getLookingAt(player, RayTraceFluidMode.ALWAYS);
+            }
+        };
+        public abstract RayTraceResult getLookingAt(EntityPlayer player);
+        public byte id;
```

Comment:
Ouch... Don't do public fields - especially non-final ones! **This should have a getter and be final**...